stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -1694,11 +1673,300 @@ fn delegate_stack_stx_signer_key() {
+        &key_to_stacks_addr(alice_stacker_key).to_account_principal(),
+    )
+    .unwrap()
+    .expect_tuple();
+
+    // Testing new signer-key correctly set
+    let state_signer_key_new = new_stacking_state.get("signer-key").unwrap();
+    assert_eq!(
+        state_signer_key_new,
+        &Value::buff_from(bob_new_signer_public_key.to_bytes_compressed()).unwrap()
+    );
+}
+
+// In this test case, Alice is a solo stacker-signer.
+//  Alice stacks the stacking minimum for two cycles.
+//  In the next cycle, Alice calls stack-increase to increase
+//  her total-locked by a second stacking minimum.
+//
+// This test asserts that Alice's total-locked is equal to
+//  twice the stacking minimum after calling stack-increase.
+#[test]
+fn stack_increase() {
+    let lock_period = 2;
+    let observer = TestEventObserver::new();
+    let (burnchain, mut peer, keys, latest_block, block_height, mut coinbase_nonce) =
+        prepare_pox4_test(function_name!(), Some(&observer));
+
+    let mut alice_nonce = 0;
+    let alice_stacking_private_key = &keys[0];
+    let alice_address = key_to_stacks_addr(alice_stacking_private_key);
+    let alice_signing_private_key = &keys[1];
+    let alice_signing_public_key = StacksPublicKey::from_private(alice_signing_private_key);
+    let min_ustx = get_stacking_minimum(&mut peer, &latest_block);
+    let pox_addr = PoxAddress::from_legacy(
+        AddressHashMode::SerializeP2PKH,
+        key_to_stacks_addr(alice_stacking_private_key).bytes,
+    );
+
+    let stack_stx = make_pox_4_lockup(
+        alice_stacking_private_key,
+        alice_nonce,
+        min_ustx,
+        pox_addr.clone(),
+        lock_period,
+        alice_signing_public_key.clone(),
+        block_height as u64,
+    );
+
+    // Initial tx arr includes a stack_stx pox_4 helper found in mod.rs
+    let txs = vec![stack_stx];
+
+    let latest_block = peer.tenure_with_txs(&txs, &mut coinbase_nonce);
+    let stacking_state = get_stacking_state_pox_4(
+        &mut peer,
+        &latest_block,
+        &key_to_stacks_addr(alice_stacking_private_key).to_account_principal(),
+    )
+    .expect("No stacking state, stack-stx failed")
+    .expect_tuple();
+
+    // Testing initial signer-key correctly set
+    let state_signer_key = stacking_state.get("signer-key").unwrap();
+    assert_eq!(
+        state_signer_key,
+        &Value::buff_from(alice_signing_public_key.to_bytes_compressed()).unwrap()
+    );
+
+    alice_nonce += 1;
+
+    let stack_increase =
+        make_pox_4_stack_increase(alice_stacking_private_key, alice_nonce, min_ustx);
+    // Next tx arr includes a stack_increase pox_4 helper found in mod.rs
+    let txs = vec![stack_increase];
+    let latest_block = peer.tenure_with_txs(&txs, &mut coinbase_nonce);
+    let stacker_transactions = get_last_block_sender_transactions(&observer, alice_address);
+
+    let actual_result = stacker_transactions
+        .first()
+        .map(|tx| tx.result.clone())
```

Comment:
Is the map necessary?

Reply:
Yes. Removing it breaks tests.

Reply:
I mean, `first()` returns an `Option`, since you are just cloning you could perhaps use `cloned()`.

Reply:
Ah, yeap, understood & updated to use .cloned() instead.